src/connectors/xiami.js

```diff
@@ -1,22 +1,17 @@
 'use strict';
 
-Connector.playerSelector = '#player-main';
+Connector.playerSelector = '.play-bar';
 
-Connector.trackSelector = '#J_trackName';
+Connector.trackSelector = '.music .info .title';
 
-Connector.albumSelector = '.ui-track-current .c3>a';
+Connector.artistSelector = '.music .info .singers';
 
-Connector.artistSelector = '#J_trackInfo>a:nth-child(2)';
+Connector.playButtonSelector = '.main-control .play-btn';
 
-Connector.playButtonSelector = '.play-btn';
+Connector.currentTimeSelector = '.audio-progress .range .bar .handle';
 
-Connector.currentTimeSelector = '#J_positionTime';
+Connector.trackArtSelector = '.music .cover-link .active img';
 
-Connector.trackArtSelector = '#J_playerCoverImg';
-
-Connector.getDuration = () => {
```

Comment:
Is there no way to get the duration of the track now?

Reply:
I removed codes 

`Connector.currentTimeSelector = '.audio-progress .range .bar .handle';`

and add these

```
Connector.getDuration = () => {
	let text = $('.audio-progress .range .bar .handle').val();
	let str = text.split("/");
	return str[0];
};
```

I removed the currentTimeSelector and added getDuration() method.
Please check :)